openlibrary/utils/isbn.py

```diff
@@ -3,7 +3,9 @@
 logger = logging.getLogger("openlibrary")
 
 def check_digit_10(isbn):
-    """Is this code right?... The length checks seem off by 1"""
+    """I believe this is checking the case where ISBNs start with a
+    leading X? It only appears to be used in isbn_13_to_isbn_10 and
+    isbn_10_to_isbn_13"""
     if len(isbn) != 9:
```

Comment:
These check_digit methods are just (re?)calculating the check digits for isbns, which is the last character.
`[1..9][check_digit]`

It might be clearer if the method was named something like `validate_check_digit_10(isbn_10)` and returned a bool since the intent appears to be confirming the check digit, not creating it.

Looking at the code below, the method should probably just be a `is_valid_isbn_10(isbn)` and return a bool after checking all the things that would make it valid.